firestore/firestore_snippets/listen_changes.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// [START fs_listen_changes]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/firestore"
+)
+
+// listenChanges listens to a query, returning the list of DocumentChange
+// in the first snapshot.
+func listenChanges(w io.Writer, projectID string) error {
+	// projectID := "project-id"
+	ctx := context.Background()
+
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	snap, err := client.Collection("cities").Where("state", "==", "CA").Snapshots(ctx).Next()
```

Comment:
`Next()` probably should be called within `for`. For now it'll only listen for a single one snapshot, but I suspect the samples intended to show the endless listening

Reply:
Looking at Python sample I'd say it registers a callback function, so on every new snapshot it'll call the callback func (cycle). In Go there is no registering, but moving the `Next()` call into `for` cycle will do the same thing. Seems to me all the `Next()` calls must be moved into `for` cycles, or it'll be listening just for a single one change

What are the local tests failures?